File: app/src/main/java/com/vktiwari/pagingretrofitrxjava/ui/peoplelist/adapter/PeopleViewHolder.java
package com.vktiwari.pagingretrofitrxjava.ui.peoplelist.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.vktiwari.pagingretrofitrxjava.R;
import com.vktiwari.pagingretrofitrxjava.model.People;

class PeopleViewHolder extends RecyclerView.ViewHolder {
    private TextView mTextView;
    private OnItemClickListener itemClickListener;

    private PeopleViewHolder(View view, OnItemClickListener itemClickListener) {
        super(view);
        this.itemClickListener = itemClickListener;
        mTextView = view.findViewById(R.id.text_name);
    }

    void bind(People people) {
        if (people != null) {
            mTextView.setText(people.name);
            mTextView.setOnClickListener(v -> itemClickListener.onClick(people));
        }

    }

    public static PeopleViewHolder create(ViewGroup parent, OnItemClickListener itemClickListener) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View view = layoutInflater.inflate(R.layout.item_people, parent, false);
        return new PeopleViewHolder(view, itemClickListener);
    }
}

File: app/src/kotlin/kotlin/com/vktiwari/pagingretrofitrxjava/ui/peoplelist/adapter/PeopleViewHolder.kt
package com.vktiwari.pagingretrofitrxjava.ui.peoplelist.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.vktiwari.pagingretrofitrxjava.R
import com.vktiwari.pagingretrofitrxjava.model.People

class PeopleViewHolder(view: View, val itemClickListener: OnItemClickListener) : RecyclerView.ViewHolder(view) {
    private var mTextView: TextView

    init {
        mTextView = view.findViewById(R.id.text_name)
    }

    fun bind(people: People?) {
        if (people != null) {
            mTextView.text = people.name
            mTextView.setOnClickListener { v -> itemClickListener.onClick(people) }
        }

    }

    companion object {
        fun create(parent: ViewGroup, itemClickListener: OnItemClickListener): PeopleViewHolder {
            val layoutInflater = LayoutInflater.from(parent.context)
            val view = layoutInflater.inflate(R.layout.item_people, parent, false)
            return PeopleViewHolder(view, itemClickListener)
        }
    }

}